core/src/main/java/de/danoeh/antennapod/core/storage/DBTasks.java

```diff
@@ -349,6 +349,10 @@ public void run() {
 
             }.start();
         }
+        // #2448: First, add to-download items to the queue before actual download
+        // so that the resulting queue order is the same as when download is clicked
+        DBWriter.addQueueItem(context, items);
```

Comment:
Could this lead to unexpected streaming of episodes while the episode is not yet downloaded or is this already fixed by your other PR about updating the queue?

Reply:
Good catch. PR #3471 fixed the issue. 

The behavior introduced here makes it more likely to encounter the bug.